pulsar-client/src/main/java/org/apache/pulsar/client/impl/schema/generic/GenericAvroSchema.java

```diff
@@ -78,4 +79,8 @@ public GenericRecord decode(byte[] bytes, byte[] schemaVersion) {
         }
     }
 
+    @Override
+    public GenericRecordBuilder newRecordBuilder() {
```

Comment:
Since the builder only handled primative types, we should throw an exception if we try to create a builder with a schema that has complex types.

Reply:
1) Technically GenericRecordBuilder doesn't limit the complex types. It is only a wrapper over AVRO builder, so whatever AVRO supports it can support. So it doesn't actually fail if you try to set complex types that AVRO supports. So I don't think we need to fail it when a schema has complex types.

2) secondly a GenericRecord schema can only be built by GenericRecordSchemaBuilder, which has thrown exception if you are using complex types. So you can't really create a generic record builder without creating a generic schema. So we are safe here.

3) I have a subsequent pull request to support complex types.

Reply:
Sure. Looking at the avro generic record builder, I don't see how that can even do complex types.